cmd/up/errors.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package up
+
+import (
+	"strings"
+
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+)
+
+// isTransient is an extension of the oktetoErrors.IsTransient, this variant is used to add transient errors dynamically
+// based on the state of the upContext, in particular, if the up session was successfully started once before any retry
+func (up *upContext) isTransient(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	isTransientErr := oktetoErrors.IsTransient(err)
+
+	if up.success {
+		if strings.Contains(err.Error(), "syncthing local=false didn't respond after") {
+			return true
+		}
+		if !isTransientErr && up.unhandledTransientRetryCount < up.unhandledTransientMaxRetries {
```

Comment:
Can we add a log here in order to track while debugging?

Reply:
I was planning to add a log, but then realized that we can infer by looking at the logs, if a non transient error is triggering a new sequence, it means it was treated as transient. But now, explaining this, it makes me realize that a log is just easier :) Will add

Reply:
log added in c488d90b

cc @jLopezbarb feel free to suggestion a different log message